lib/utils/message_converter.dart

```diff
@@ -526,10 +526,10 @@ class MessageConverter {
         );
       }
 
-      // If not found in contacts, try metadata cache with build-safe scheduling
+      // If not found in contacts, fetch directly from API with build-safe scheduling
       return await Future.microtask(() async {
-        final metadataCache = ref.read(metadataCacheProvider.notifier);
-        final contactModel = await metadataCache.getContactModel(pubkey);
+        final userProfileNotifier = ref.read(userProfileDataProvider.notifier);
```

Comment:
I'm avoiding this because this triggers a N+1. This will query de local db  to get the metadata  once per message, cause user profile data provider is calling the rust method. That was my first approach to finally remove the metadata cache providers, but I think this will break and make slow chats with more messages. Please leave the metadata cache provider for now, the real fix is coming soon (which is to load the metadata of group members not at message level but at group level and then pass that to the message converter).